### ISSUE TITLE
[roctracer] error: ‘int pthread_yield()’ is deprecated: pthread_yield is deprecated, use sched_yield instead

### ISSUE THREAD

**Issue by jaschaknack** [rocm-arch/rocm-arch#688]
Installation of roctracer fails with error. I believe the following bug is related and they have patched it. [https://bugs.gentoo.org/808049](https://bugs.gentoo.org/808049)

```
make[2]: Leaving directory '/tmp/makepkg/roctracer/src/build'
[ 69%] Built target ctrl
In file included from /tmp/makepkg/roctracer/src/roctracer-rocm-5.0.0/src/core/roctracer.cpp:44:
/tmp/makepkg/roctracer/src/roctracer-rocm-5.0.0/src/core/memory_pool.h: In static member function ‘static void roctracer::MemoryPool::sync_reader(const roctracer::MemoryPool::consumer_arg_t*)’:
/tmp/makepkg/roctracer/src/roctracer-rocm-5.0.0/src/core/memory_pool.h:155:64: error: ‘int pthread_yield()’ is deprecated: pthread_yield is deprecated, use sched_yield instead [-Werror=deprecated-declarations]
  155 |     while(arg->valid.load() == true) PTHREAD_CALL(pthread_yield());
      |                                                   ~~~~~~~~~~~~~^~
/tmp/makepkg/roctracer/src/roctracer-rocm-5.0.0/src/core/memory_pool.h:36:15: note: in definition of macro ‘PTHREAD_CALL’
   36 |     int err = call;                                                                                \
      |               ^~~~
In file included from /usr/include/features.h:490,
                 from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdint.h:26,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/include/stdint.h:9,
                 from /tmp/makepkg/roctracer/src/roctracer-rocm-5.0.0/inc/roctracer.h:39,
                 from /tmp/makepkg/roctracer/src/roctracer-rocm-5.0.0/src/core/roctracer.cpp:23:
/usr/include/pthread.h:479:12: note: declared here
  479 | extern int __REDIRECT_NTH (pthread_yield, (void), sched_yield)
      |            ^~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/roctracer64.dir/build.make:79: CMakeFiles/roctracer64.dir/src/core/roctracer.cpp.o] Error 1
make[2]: Leaving directory '/tmp/makepkg/roctracer/src/build'
make[1]: *** [CMakeFiles/Makefile2:139: CMakeFiles/roctracer64.dir/all] Error 2
make[1]: Leaving directory '/tmp/makepkg/roctracer/src/build'
make: *** [Makefile:159: all] Error 2
make: Leaving directory '/tmp/makepkg/roctracer/src/build'
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: roctracer
```

**Comment by tpkessler** [comment]
Thanks for reporting and for the link to the patch @jaschaknack! I've updated the package to 5.0.2.

**Comment by tpkessler** [comment]
Patched with e1f3e08eaa0512422d0743bcdd6f730dcafe03df

**Comment by acxz** [comment]
Closed with https://github.com/rocm-arch/rocm-arch/commit/a421cdc25ab95c2e6c053a33e4779672162d9f97
